docs-src/concepts.md

```diff
@@ -61,18 +61,52 @@ model.
 > Custom Resource Definitions (CRDs). Unqualified resource names will implicitly
 > be assumed to be part of this API group.
 
-Our resource model is based around a separation of concerns for a service
-producer. Each resource is intended to be (mostly) independently-evolvable and
-self-consistent:
-
-* A way to group pods (backends) into a set via label selection: Kubernetes
-  `core.Service`, independent of application-level routing.
-* A way to describe application-level routing: `xxxxRoute`, e.g. `HTTPRoute`,
-  `TCPRoute` independent of traffic access and consumption.
-* A way to describe traffic access and consumption: `Gateway`, independent of
-  implementation.
-* A way to describe which implementations of traffic access are available:
-  `GatewayClass`.
+
+There are three main types of object in our resource model:
+
+*GatewayClass* defines a set of gateways with a common configuration and behavior. 
+
+*Gateway* requests a point where traffic can be translated to Services within the cluster.
+
+*Routes* describe how traffic coming via the Gateway maps to the Services.
+
+### GatewayClass
+
+GatewayClass defines a set of Gateways that share a common configuration and behaviour.
+Each GatewayClass will be handled by a single controller, although controllers MAY handle more than one.
+
+GatewayClass is a cluster-scoped resource.
+There MUST be at least one GatewayClass defined in order to be able to have functional Gateways.
+A controller that implements the Gateway API does so by providing an associated GatewayClass resource that the user can reference from their Gateway(s).
+
+This is similar to IngressClass for Ingress and StorageClass for PersistentVolumes.
+In Ingress v1beta1, the closest analog to GatewayClass is the `ingress-class` annotation, and in IngressV1, the closest analog is the IngressClass object.
+
+### Gateway
+
+A Gateway describes how traffic can be translated to Services within the cluster.
+That is, it defines a request for a way to translate traffic from somewhere that does not know about Kubernetes to somewhere that does.
+For example: HTTP/S traffic sent to K8S Services via the Gateway, which may be a Cloud LB, in-cluster proxy or external hardware LB.
+While many use cases have client traffic originating “outside” the cluster, this is not a strict requirement. 
+
+It defines a request for a specific load balancer config that implements the GatewayClass’ configuration and behaviour contract.
+The resource MAY be created by an operator directly, or MAY be created by a controller handling a GatewayClass.
+
+A valid Gateway spec SHOULD provide a Listener definition that MAY request addresses and ports, protocol termination, and TLS settings. Listener definitions MAY also be inherited from GatewayClass defaults.
```

Comment:
I think this sentence probably doesn't belong here but in an implementors guide? It references specific fields in the API such as Listener that the reader will have to forward reference when reading.

How about:

```
As the Gateway spec captures user intent, it may not contain a complete specification
for all attributes in the spec. For example, the user may omit fields such as addresses,
ports, TLS settings. This allows the controller managing the GatewayClass to provide 
these settings for the user, resulting in a more portable spec.
```